src/Workspaces/Core/Portable/Formatting/Formatter.cs

```diff
@@ -246,10 +246,13 @@ internal static IFormattingResult GetFormattingResult(SyntaxNode node, IEnumerab
                 return null;
             }
 
+            var optionService = workspace.Services.GetRequiredService<IOptionService>();
+
             options ??= workspace.Options;
             rules ??= GetDefaultFormattingRules(workspace, node.Language);
             spans ??= SpecializedCollections.SingletonEnumerable(node.FullSpan);
-            return languageFormatter.Format(node, spans, options, rules, cancellationToken);
+            var shouldUseFormattingSpanCollapse = options.GetOption(FormattingOptions.AllowDisjointSpanMerging);
+            return languageFormatter.Format(node, spans, shouldUseFormattingSpanCollapse, options.AsAnalyzerConfigOptions(optionService, node.Language), rules, cancellationToken);
```

Comment:
Why this change of explicitly querying `FormattingOptions.AllowDisjointSpanMerging` and pass it in? I presume invoking GetOption on the passed in AnalyzerConfigOptions argument should give the same value to the callee given that the wrapping AnalyzerConfigOptions implementation falls back to the original option set?

Reply:
`AllowDisjointSpanMerging` isn't backed by **.editorconfig**, so there is no way to provide it through `AnalyzerConfigOptions`.